src/utils/utils.cairo

```diff
@@ -207,47 +207,23 @@ namespace Helpers {
     // @return The data read from calldata
     func slice_data{range_check_ptr}(
         data_len: felt, data: felt*, data_offset: felt, slice_len: felt
-    ) -> felt* {
-        alloc_locals;
-        let (local sliced_data: felt*) = alloc();
-
-        let diff = data_len - data_offset;
-
-        let is_diff_greater_than_element_len: felt = is_le_felt(slice_len, diff);
-
-        if (is_diff_greater_than_element_len == 0) {
-            memcpy(dst=sliced_data, src=data + data_offset, len=diff);
-
-            let pad_n: felt = slice_len - diff;
-
-            fill(arr=sliced_data + diff, value=0, length=pad_n);
-        } else {
-            memcpy(dst=sliced_data, src=data + data_offset, len=slice_len);
-        }
-
-        return (sliced_data);
-    }
-
-    // @notice Returns a zero-padded slice of an array
-    // @param array_len Length of the array
-    // @param array Pointer to the array
-    // @param slice_len Number of elements to take
-    // @param offset Take elements in the array starting from this position
-    // @return Pointer to the zero-padded copy of the array slice
-    func array_slice{range_check_ptr}(
-        array_len: felt, array: felt*, slice_len: felt, offset: felt
     ) -> felt* {
         alloc_locals;
         local len: felt;
-        let (local new_array: felt*) = alloc();
+        let (local new_data: felt*) = alloc();
+
+        // slice's len = min(slice_len, data_len-offset, 0)
+        // which corresponds to full, partial or empty overlap with data
+        // The result is zero-padded in case of partial or empty overlap.
 
-        let is_non_empty: felt = is_le_felt(offset, array_len);
-        let max_len: felt = (array_len - offset) * is_non_empty;
+        let is_non_empty: felt = is_le_felt(data_offset, data_len);
+        let max_len: felt = (data_len - data_offset) * is_non_empty;
         let is_within_bound: felt = is_le_felt(slice_len, max_len);
         let len = max_len + (slice_len - max_len) * is_within_bound;
-        memcpy(dst=new_array, src=array + offset, len=len);
-        fill(arr=new_array + len, value=0, length=slice_len - len);
-        return new_array;
+
+        memcpy(dst=new_data, src=data + data_offset, len=len);
+        fill(arr=new_data + len, value=0, length=slice_len - len);
```

Comment:
So for an out of bounds offset, with slice_len = x, this function will return an array of length x, filled with zeros?

Reply:
Yes! Offset out of bounds means all 0's. I am not sure if there are some opcodes for which the zero-padding rule is not applicable, but on a first look, it seems to be consistent through out the EVM. Arguably, we could separate slicing and padding if ever needed.